glue/core/data_factories/npy.py

```diff
@@ -42,6 +41,22 @@ def npy_reader(filename, format='auto', auto_merge=False, **kwargs):
 
     return d
```

Comment:
A simpler solution, rather than converting the plain array to an unstructured array, would be to do something more along the lines of:

```python
d = Data()
if npy_data.dtype.names is None:
    comp = Component.autotyped(npy_data)
    d.add_component(comp, label='array')
else:
    for name in npy_data.dtype.names:
         comp = Component.autotyped(npy_data[name])
         d.add_component(comp, label=name)
```